app/routes.py

```diff
@@ -1 +1,254 @@
+                "id":new_goal.goal_id,
+                "title":new_goal.title
+            }
+        }, 201
+
+    elif request.method == "GET":
+        sorting_goals= request.args.get('sort') 
+        list = None
+        if sorting_goals== "desc":
+            list = Goal.query.order_by(Goal.title.desc()) # descending method
+        elif sorting_goals == "asc":
+            list = Goal.query.order_by(Goal.title.asc()) # ascending method
+        else: 
+            list = Goal.query.all()
+        goals_response = []
+        for goal in list:
+            goals_response.append({
+            "id": goal.goal_id,
+            "title": goal.title,
+            })
+        
+        return jsonify(goals_response), 200
+
+@goals_bp.route("/<goal_id>", methods=["GET","PUT","DELETE"])
+def handle_goal_get(goal_id):
+    goal = Goal.query.get(goal_id)
+    if goal == None:
+        return ("", 404)
+
+    if request.method == "GET":
+        return {
+            "goal": {
+                "id":goal.goal_id,
+                "title":goal.title,          
+                }
+        }
+    if request.method == "PUT":
+        form_data = request.get_json()
+
+        goal.title = form_data["title"]
+
+        db.session.commit()
+
+        return jsonify({
+            "goal":{
+                "id":goal.goal_id,
+                "title":goal.title,
+            }
+        }),200
+
+    elif request.method == "DELETE":
+        db.session.delete(goal)
+        db.session.commit()
+
+        if not goal in Goal.query:
+            abort(make_response({"message": f"Goal {goal_id} not found"}, 404))
+        return jsonify({
+            "details": f'Goal {goal.goal_id} "{goal.title}" successfully deleted'
+        }),200
+
+
+@goals_bp.route("/<goal_id>/tasks", methods=["POST","GET"])
+def post_tasked_goal(goal_id):
+
+        goal = Goal.query.get(goal_id)
+
+        if goal == None:
+            return (""), 404
+
+        if request.method == "POST":
+            request_body = request.get_json()
+
+            tasks_instances= []
+            for task_id in request_body["task_ids"]:
+                tasks_instances.append(Task.query.get(task_id))
+
+            goal.tasks = tasks_instances
+
+            db.session.commit()
+
+            task_ids = []
+            for task in goal.tasks:
+                task_ids.append(task.task_id)
+
+            response_body = {
+                        "id": goal.goal_id,
+                        "task_ids": task_ids
+                    }
+
+            return jsonify(response_body), 200
+
+        if request.method == "GET":
+            tasks_response =[]
+            for task in goal.tasks:
+                tasks_response.append({
+                    "id": task.task_id,
+                    "goal_id": task.goal_id,
+                    "title": task.title,
+                    "description": task.description,
+                    "is_complete": bool(task.completed_at)
+                })
```

Comment:
This looks nearly identical to what you have in the instance method to_dict in the Task class. We should use the to_dict method on each task from goal.tasks instead of repeating code. You already have the logic to handle adding goal_id to each task dict too in to_dict!